kubernetes/structures_deployment.go

```diff
@@ -14,12 +14,26 @@ func flattenDeploymentSpec(in appsv1.DeploymentSpec) ([]interface{}, error) {
 		att["replicas"] = *in.Replicas
 	}
 
-	att["selector"] = in.Selector
+	if in.ProgressDeadlineSeconds != nil {
+		att["progress_deadline_seconds"] = *in.ProgressDeadlineSeconds
+	}
+
+	if in.RevisionHistoryLimit != nil {
+		att["revision_history_limit"] = *in.RevisionHistoryLimit
+	}
+
+	if in.Selector != nil {
+		att["selector"] = flattenLabelSelector(in.Selector)
+	}
+
 	podSpec, err := flattenPodSpec(in.Template.Spec)
 	if err != nil {
 		return nil, err
 	}
-	att["template"] = podSpec
+	template := make(map[string]interface{})
+	template["spec"] = podSpec
+	template["metadata"] = flattenMetadata(in.Template.ObjectMeta)
```

Comment:
there's a crash here if `in.Template` is nil here?

Reply:
Template attribute is not a pointer.
This API object is a bit messy, because some attributes are actually pointers and some are not.